tests/components/daikin/test_config_flow.py

```diff
@@ -27,13 +27,13 @@ def init_config_flow(hass):
 def mock_daikin():
     """Mock pydaikin."""
 
-    async def mock_daikin_init():
+    async def mock_daikin_factory(*args, **kwargs):
         """Mock the init function in pydaikin."""
-        pass
+        return Appliance
 
     with patch("homeassistant.components.daikin.config_flow.Appliance") as Appliance:
-        Appliance().values.get.return_value = "AABBCCDDEEFF"
-        Appliance().init = mock_daikin_init
+        type(Appliance).mac = PropertyMock(return_value="AABBCCDDEEFF")
+        Appliance.factory = mock_daikin_factory
```

Comment:
@MartinHjelmare this was enough, but your suggestion put me in the right direction.

```suggestion
        Appliance.factory.side_effect = mock_daikin_factory
```